.github/workflows/test.yml

```diff
@@ -66,14 +58,34 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
+      - name: Find and merge benchmarks
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        run: jq -s '.[0].benchmarks = ([.[].benchmarks] | add) | if .[0].benchmarks == null then null else .[0] end' opentelemetry-*/tests/*${{ matrix.package }}*.json > output.json
```

Comment:
What is this code doing? Is `.json` outputs in the test folders a result from the previous step?

Reply:
This is the solution that we came up with @aabmass . The problem is that tests run as `tox -f core` and so it matches and runs tests for any package with "core" in it `tox.ini`. The benchmark tests would be split across all the core packages (i.e. `core-sdk` `core-api` `core-instruemntation` etc.) and we wouldn't be able to give just 1 file to the next action to automate performance tests.

So we use this `jq` command to combine the benchmarks for the tests under a single `core`/Python Version benchmark test file so that we can work with the subsequent action which requires only 1 file.

Reply:
The output file is called `output.json` which the subsequent steps needs. How come `output-file-path` for the subsequent action is also `output.json`?

Reply:
Ah sorry if that's confusing, the subsequent `output.json` _is_ the input to the action. It has to have the same name so that the subsequent action know where to find it.

So `Find and merge benchmarks` creates a file called `output.json`.

And `Report on benchmark results` uses the file `output.json` that was created in the pervious step.

Reply:
Yeah they makes sense to me. " output-file-path" is just kind of a confusing name.

Reply:
Yes unfortunately 😞  See this [Minimal Setup](https://github.com/rhysd/github-action-benchmark#minimal-setup) where they have `output-file-path: output.txt`